src/detect-byte-extract.c

```diff
@@ -38,27 +34,24 @@
 #include "detect-isdataat.h"
 #include "detect-engine-build.h"
 
-#include "app-layer-protos.h"
+#include "rust.h"
 
-#include "flow.h"
-#include "flow-var.h"
-#include "flow-util.h"
+#include "app-layer-protos.h"
 
 #include "util-byte.h"
 #include "util-debug.h"
 #include "util-unittest.h"
 #include "util-unittest-helper.h"
-#include "util-spm.h"
 
 /* the default value of endianness to be used, if none's specified */
-#define DETECT_BYTE_EXTRACT_ENDIAN_DEFAULT DETECT_BYTE_EXTRACT_ENDIAN_BIG
+#define DETECT_BYTE_EXTRACT_ENDIAN_DEFAULT (uint8_t) BigEndian
 
 /* the base to be used if string mode is specified.  These options would be
  * specified in DetectByteParseData->base */
 #define DETECT_BYTE_EXTRACT_BASE_NONE 0
-#define DETECT_BYTE_EXTRACT_BASE_HEX  16
-#define DETECT_BYTE_EXTRACT_BASE_DEC  10
-#define DETECT_BYTE_EXTRACT_BASE_OCT   8
+#define DETECT_BYTE_EXTRACT_BASE_HEX  (uint8_t) BaseHex
```

Comment:
From a Rust enum as well?  I've played with giving this a prefix, but it means the Rust names and the C names are different, breaking stuff like git grep, but it also puts these names into the export C namespace which isn't good either.  But really outside the scope of this PR. Just really seeing it now though.

Reply:
Yes --- also from `rust/src/detect/lib.rs`